external/builder/babel-plugin-pdfjs-preprocessor.mjs

```diff
@@ -0,0 +1,241 @@
+import { types as t, transformSync } from "@babel/core";
+import fs from "fs";
+import { join as joinPaths } from "path";
+import vm from "vm";
+
+const PDFJS_PREPROCESSOR_NAME = "PDFJSDev";
+const ROOT_PREFIX = "$ROOT/";
+
+function isPDFJSPreprocessor(obj) {
+  return obj.type === "Identifier" && obj.name === PDFJS_PREPROCESSOR_NAME;
+}
+
+function evalWithDefines(code, defines) {
+  if (!code || !code.trim()) {
+    throw new Error("No JavaScript expression given");
+  }
+  return vm.runInNewContext(code, defines, { displayErrors: false });
+}
+
+function handlePreprocessorAction(ctx, actionName, args, path) {
+  try {
+    const arg = args[0];
+    switch (actionName) {
+      case "test":
+        if (!t.isStringLiteral(arg)) {
+          throw new Error("No code for testing is given");
+        }
+        return !!evalWithDefines(arg.value, ctx.defines);
+      case "eval":
+        if (!t.isStringLiteral(arg)) {
+          throw new Error("No code for eval is given");
+        }
+        const result = evalWithDefines(arg.value, ctx.defines);
+        if (
+          typeof result === "boolean" ||
+          typeof result === "string" ||
+          typeof result === "number" ||
+          typeof result === "object"
+        ) {
+          return result;
+        }
+        break;
+      case "json":
+        if (!t.isStringLiteral(arg)) {
+          throw new Error("Path to JSON is not provided");
+        }
+        let jsonPath = arg.value;
+        if (jsonPath.startsWith(ROOT_PREFIX)) {
+          jsonPath = joinPaths(
+            ctx.rootPath,
+            jsonPath.substring(ROOT_PREFIX.length)
+          );
+        }
+        return JSON.parse(fs.readFileSync(jsonPath, "utf8"));
+    }
+    throw new Error("Unsupported action");
+  } catch (e) {
+    throw path.buildCodeFrameError(
+      "Could not process " +
+        PDFJS_PREPROCESSOR_NAME +
+        "." +
+        actionName +
+        ": " +
+        e.message
+    );
+  }
+}
+
+function babelPluginPDFJSPreprocessor(babel, ctx) {
+  return {
+    name: "babel-plugin-pdfjs-preprocessor",
+    manipulateOptions({ parserOpts }) {
+      parserOpts.attachComment = false;
+    },
+    visitor: {
+      "ExportNamedDeclaration|ImportDeclaration": ({ node }) => {
+        if (node.source && ctx.map?.[node.source.value]) {
+          node.source.value = ctx.map[node.source.value];
+        }
+      },
+      "IfStatement|ConditionalExpression": {
+        exit(path) {
+          const { node } = path;
+          if (t.isBooleanLiteral(node.test)) {
+            // if (true) stmt1; => stmt1
+            // if (false) stmt1; else stmt2; => stmt2
+            path.replaceWith(
+              node.test.value === true
+                ? node.consequent
+                : node.alternate || t.emptyStatement()
+            );
+          }
+        },
+      },
+      UnaryExpression(path) {
```

Comment:
The fix is probably to add `exit` here like to the other visitors (I can submit a PR tomorrow, I'm from my phone right now), so that the UnaryExpression is checked after that the inner PDFJSDev.test call is replaced.